src/use-popable/types.ts

```diff
@@ -0,0 +1,4 @@
+export type PopableManager = {
+  hide: () => void;
+  show: () => void;
+};
```

Comment:
I understand this is clean, but I doubt `usePopable` will evolve that much in the future.

I'd rather keep it all in one file (unless I'm missing the reason why it shouldn't?), also because that's how the other files have been structured so far.

Just let me know if there was another reason for your decision, otherwise just having `src/use-popable.ts` would be simpler

Reply:
Yeah, understandable. However, having both default-export component in the same file as a named export can break Fast Refresh. I've learned this the hard way, so I opt to make them separate files. And since both `use-popable` and `Popable` import `PopableManager`, I put this type in a separate file to avoid circular imports.

See: https://nextjs.org/docs/basic-features/fast-refresh#limitations

<img width="756" alt="Screen Shot 2021-04-15 at 11 52 01 AM" src="https://user-images.githubusercontent.com/13172299/114899521-07438500-9de1-11eb-8305-b4b97efdb153.png">

Reply:
Great, thank you for clarifying. Then that's even better :)